Filter venues based on production selections

randycoulman opened the pull request:
When selecting one or more productions for an item, restrict the venue selection list to only those venues associated with the selected productions.

For now, we do this by disabling any other venues.  With chosen.js, those disabled venues will be in the list, but unselectable.  We can change this behavior with an option if we decide to do that later.

randycoulman committed:
Filter venues based on selected productions

This is basically working; need some specs on the new VenuesController

randycoulman committed:
Add specs for venues controller

randycoulman committed:
Add a model spec for the new venue scope

seanculver commented on vendor/engines/production_credits/app/controllers/production_credits/venues_controller.rb:
```diff
@@ -0,0 +1,12 @@
+    skip_before_filter :authenticate_user!
+
+    def index
+      venues = Venue.for_production_ids(params[:production_ids])
```
you could skip this step if you already know there are no production_ids

```
venues = Venue.for_production_ids(params[:production_ids]) if params[:production_ids]
```

Or some variation of that conditional.

randycoulman replied:
Good idea!  Addressed in 5200483c6cefc5ec15fe18221bab0db17c5366a5.

randycoulman committed:
Don't request filtered venues when no production_ids provided

randycoulman committed:
Add faker to Gemfile

Faker is only a development dependency of the production_credits
engine, so it doesn’t get pulled in automatically.

The test failure made me realize that the engine factories are
available in the main app, so I refactored one spec to make use of the
factory.

seanculver commented:
:+1:
